app/assets/javascripts/scripts.js

```diff
@@ -1,32 +1,79 @@
-    }
+  let cancel = confirm("Are you sure you want to cancel creating this " + input + "? The information will not be saved!");
+  if (cancel) {
+    window.location.href = "/" + input + "s"
+  }
+}
+
+function addExistingTag() {
+  let parent = $(event.target).parent();
+  let selectedOption = parent.find("select option:selected");
+  let tagName = selectedOption.text();
+  let tagId = selectedOption.val();
+  addNewSelect(tagName, tagId);
 }
 
 function addNewTag() {
-    var newTagName = $("#new_tag_field").val();
-    var newTagNum = $(".tag").length;
-    var namePrefix = "vendor[tags_attributes][" + newTagNum + "]";
-    var idPrefix = "vendor_tags_attributes_" + newTagNum + "_";
-
-    var newDiv = $("<div>").attr({class: "tag"});
-
-    $("<label>").attr({for: idPrefix + "name"}).text("Name").appendTo(newDiv);
-    $("<input>").attr({type: "text", value: newTagName, name: namePrefix + "[name]", id: idPrefix + "name"}).appendTo(newDiv);
-    $("<input>").attr({type: "hidden", value: "0", name: namePrefix + "[_destroy]"}).appendTo(newDiv);
-    $("<input>").attr({type: "checkbox", value: "1", name: namePrefix + "[_destroy]", id: idPrefix + "_destroy"}).appendTo(newDiv);
-    $("<label>").attr({for: idPrefix + "_destroy"}).text("Remove Tag").appendTo(newDiv);
-    newDiv.appendTo("#tags");
-}
\ No newline at end of file
+  let parent = $(event.target).parent();
+  let tagName = parent.find('input[type="text"]').val();
+  addNewSelect(tagName, null);
+}
+
+function addNewSelect(tagName, tagId) {
+  let parent = $(event.target).parent();
+  let tagTypePlural = parent.attr("id").slice(4);
+  let tagType = tagTypePlural.slice(0, -1);
+  let tagsDiv = $(`#${tagTypePlural}`);
+  let newTagNum = $("." + tagType).length;
+
+  let formFor = $(document).find("form").attr("class").split("_")[1];  // Value is vendor or product
+  let namePrefix = `${formFor}[${tagTypePlural}_attributes][${newTagNum}]`;
+  let idPrefix = `${formFor}_${tagTypePlural}_attributes_${newTagNum}_`;
+
+  let newDiv = $("<div>").attr({class: tagType});
+  $("<input>").attr({size: 41, type: "text", value: tagName, name: namePrefix + "[name]", id: idPrefix + "name"}).appendTo(newDiv);
+  $("<input>").attr({name: namePrefix + "[_destroy]", type: "hidden", value: 0}).appendTo(newDiv);
+  $("<input>").attr({type: "checkbox", value: 1, name: namePrefix + "[_destroy]", id: idPrefix + "_destroy"}).appendTo(newDiv);
+  $("<label>").attr({for: idPrefix + "_destroy"}).text("Remove " + tagType.charAt(0).toUpperCase() + tagType.slice(1) + " Type").appendTo(newDiv);
+  newDiv.appendTo(tagsDiv);
+
+  if (tagId) {
+    // Append to nested attributes
+    $("<input>").attr({type: "hidden", value: tagId, name: namePrefix + "[id]", id: idPrefix + "id"}).insertAfter(newDiv);
+    // Append ot list of tag ids (so we can add additional existing tags)
+    $("<option>").attr({value: tagId, selected: "selected"}).appendTo($(`#${formFor}_${tagType}_ids`));
+  }
+}
+
+// function addNewSelect() {
```

Comment:
Is there a reason you’re leaving this code in? If it’s not going to be used it should be deleted to keep the code clean.

Reply:
This was also from a previous attempt, so I've deleted the commented code.